src/main/java/fi/metatavu/metaform/server/rest/MetaformsApiImpl.java

```diff
@@ -296,11 +313,48 @@ public Response listReplies(UUID metaformId, UUID userId, String createdBeforePa
         modifiedAfter,
         includeRevisions != null && includeRevisions,
         fieldFilters);
-    
+    replies.forEach(reply -> auditLogEntryController.createAuditLogEntry(getLoggerUserId(), AuditLogEntryType.VIEW_REPLY,
+            reply.getId(), null, reply.getId()+" was viewed by user "+getLoggerUserId()));
     List<Reply> result = getPermittedReplies(replies, AuthorizationScope.REPLY_VIEW).stream()
       .map(entity -> replyTranslator.translateReply(metaformEntity, entity, null))
       .collect(Collectors.toList());
-    
+
+    return createOk(result);
+  }
+
+  @Override
+  public Response findAuditLogEntries(UUID metaformId, UUID userId, String createdBeforeParam, String createdAfterParam, String modifiedBeforeParam, String modifiedAfterParam) {
+    if (!hasRealmRole(VIEW_AUDIT_LOGS_ROLE)) {
+      return createForbidden("Only users with "+VIEW_AUDIT_LOGS_ROLE+" can access this view");
+    }
+    OffsetDateTime createdBefore = parseTime(createdBeforeParam);
+    OffsetDateTime createdAfter = parseTime(createdAfterParam);
+    OffsetDateTime modifiedBefore = parseTime(modifiedBeforeParam);
+    OffsetDateTime modifiedAfter = parseTime(modifiedAfterParam);
+
+    if (!isRealmUser()) {
```

Comment:
Isn't this useless? User is already required to have preper role so it has to be realm user, right?